ratis-common/src/main/java/org/apache/ratis/util/Daemon.java

```diff
@@ -17,24 +17,59 @@
  */
 package org.apache.ratis.util;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
 public class Daemon extends Thread {
+  static final Logger LOG = LoggerFactory.getLogger(Daemon.class);
+  public static final Thread.UncaughtExceptionHandler LOG_EXCEPTION =
+      (t, e) -> LOG.error(t.getName() + " threw an uncaught exception", e);
```

Comment:
Move them to MiniRaftCluster since they are only used there.

Reply:
I feel the uncaught exceptions should default to be printed so at least users can be aware of issues and fix them. Otherwise it's really hard to catch silent crashes. I figure logging is a mild action and doesn't change the existing behavior. WDYT?

Reply:
Sounds good. This has been removed.